src/deploy/functions/cache/applyHash.ts

```diff
@@ -34,13 +36,28 @@ export async function applyBackendHashToBackends(
 function applyBackendHashToEndpoints(
   wantBackend: Backend,
   envHash: string,
+  codebaseFilters: EndpointFilter[],
   sourceV1Hash?: string,
   sourceV2Hash?: string
 ): void {
   for (const endpoint of allEndpoints(wantBackend)) {
     const secretsHash = getSecretsHash(endpoint);
     const isV2 = endpoint.platform === "gcfv2";
     const sourceHash = isV2 ? sourceV2Hash : sourceV1Hash;
-    endpoint.hash = getEndpointHash(sourceHash, envHash, secretsHash);
+    // If the endpoint is in the filtered list, then skip setting a hash (effectively forcing a deploy).
+    if (!isEndpointFiltered(endpoint.id, codebaseFilters)) {
+      endpoint.hash = getEndpointHash(sourceHash, envHash, secretsHash);
+    }
   }
 }
+
+function isCodebaseFiltered(codebase: string, filters: EndpointFilter[]) {
+  return filters.some(
+    (filter) =>
+      (!filter?.idChunks || filter?.idChunks?.length === 0) && filter.codebase === codebase
+  );
+}
+
+function isEndpointFiltered(id: string, filters: EndpointFilter[]) {
+  return filters.some((filter) => filter?.idChunks?.some((filterId) => filterId === id));
+}
```

Comment:
I think you might want to juse use `endpointMatchesFilter` function.

There is a bit of nuance here w/ grouped function (https://firebase.google.com/docs/functions/organize-functions) that's a bit complicated, and this helper doesn't correctly compensate grouped function filters.

Reply:
🙇  !!! You're absolutely right! Thank you for catching this!